sqlmesh/core/linter/definition.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+
+from sqlmesh.core.config.linter import LinterConfig
+
+from sqlmesh.core.model import Model
+
+from sqlmesh.utils.errors import raise_config_error
+from sqlmesh.core.console import get_console
+from sqlmesh.core.linter.rule import RuleSet
+
+from sqlmesh.core.config.linter import RuleListType
+
+
+def gather_rules(all_rules: RuleSet, rule_names: RuleListType) -> RuleSet:
+    if rule_names == "ALL":
+        rule_names = list(all_rules.keys())
+
+    rules = set()
+    for rule_name in rule_names:
+        if rule_name not in all_rules:
+            raise_config_error(f"Rule {rule_name} could not be found")
+
+        rules.add(all_rules[rule_name])
+
+    return RuleSet(rules)
+
+
+class Linter:
+    def fill_from_config(self, all_rules: RuleSet, config: LinterConfig) -> Linter:
```

Comment:
this should be a class method, which should initialize Linter with the appropriate arguments, where did your initializer go? 

it's not good practice to set these properties from arbitrary methods